File: _more/compiler/py2rs/test/complex_list.py
# Test with complex list comprehension and filter

numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

# List comprehension with condition
even_squares = [x**2 for x in numbers if x % 2 == 0]
print("even_squares:", even_squares)

# Filter and map
filtered = list(filter(lambda x: x > 5, numbers))
print("filtered:", filtered)

# Nested list
matrix = [[i*3+j for j in range(3)] for i in range(3)]
print("matrix:", matrix)

File: _more/compiler/py2rs/test/complex_list.rs
fn main() {
    let numbers = vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

    // List comprehension with condition
    let even_squares: Vec<i32> = numbers.iter()
        .filter(|x| *x % 2 == 0)
        .map(|x| x * x)
        .collect();
    println!("even_squares:{:?}", even_squares);

    // Filter and map
    let filtered: Vec<i32> = numbers.iter()
        .filter(|x| *x > 5)
        .cloned()
        .collect();
    println!("filtered:{:?}", filtered);

    // Nested list
    let matrix: Vec<Vec<i32>> = (0..3)
        .map(|i| {
            (0..3)
                .map(|j| i * 3 + j)
                .collect()
        })
        .collect();
    println!("matrix:{:?}", matrix);
}